mono/profiler/proflog.c

```diff
@@ -4352,7 +4359,7 @@ handle_writer_queue_entry (MonoProfiler *prof)
 
 		dump_buffer (prof, entry->buffer);
 
-		free (entry);
+		mono_thread_hazardous_try_free (entry, free_lfq_node);
```

Comment:
Unless you operate on the entry hazardously, freeing it with HP makes little sense.